src/axom/quest/examples/quest_distributed_distance_query_example.cpp

```diff
@@ -299,173 +417,238 @@ struct BlueprintParticleMesh
+    for(int i = 0; i < ptCount; ++i)
+    {
+      pts[i][0] = xs[i];
+    }
+    for(int i = 0; i < ptCount; ++i)
+    {
+      pts[i][1] = ys[i];
+    }
+    if(NDIMS == 3)
+    {
+      for(int i = 0; i < ptCount; ++i)
+      {
+        pts[i][0] = zs[i];
+      }
+    }
+    return pts;
+  }
+
   template <typename T>
   void registerNodalScalarField(const std::string& fieldName)
   {
-    SLIC_ASSERT_MSG(hasPoints(),
-                    "Cannot register a field with the BlueprintParticleMesh "
-                    "before adding points");
-
-    auto* fld = m_fieldsGroup->createGroup(fieldName);
-    fld->createViewString("association", "vertex");
-    fld->createViewString("topology", m_topoGroup->getName());
-    fld->createViewAndAllocate("values",
-                               sidre::detail::SidreTT<T>::id,
-                               numPoints());
+    for(axom::IndexType dIdx = 0; dIdx < domain_count(); ++dIdx)
+    {
+      auto* fld = m_fieldsGroups[dIdx]->createGroup(fieldName);
+      fld->createViewString("association", "vertex");
+      fld->createViewString("topology", m_topoGroups[dIdx]->getName());
+      fld->createViewAndAllocate("values",
+                                 sidre::detail::SidreTT<T>::id,
+                                 numPoints(dIdx));
+    }
   }
 
   template <typename T>
   void registerNodalVectorField(const std::string& fieldName)
   {
-    SLIC_ASSERT_MSG(hasPoints(),
-                    "Cannot register a field with the BlueprintParticleMesh "
-                    "before adding points");
-
-    const int SZ = numPoints();
     const int DIM = dimension();
+    for(axom::IndexType dIdx = 0; dIdx < domain_count(); ++dIdx)
+    {
+      const int SZ = numPoints(dIdx);
 
-    auto* fld = m_fieldsGroup->createGroup(fieldName);
-    fld->createViewString("association", "vertex");
-    fld->createViewString("topology", m_topoGroup->getName());
+      auto* fld = m_fieldsGroups[dIdx]->createGroup(fieldName);
+      fld->createViewString("association", "vertex");
+      fld->createViewString("topology", m_topoGroups[dIdx]->getName());
 
-    // create views into a shared buffer for the coordinates, with stride NDIMS
-    auto* buf = m_group->getDataStore()
-                  ->createBuffer(sidre::detail::SidreTT<T>::id, DIM * SZ)
-                  ->allocate();
-    switch(DIM)
-    {
-    case 3:
-      fld->createView("values/x")->attachBuffer(buf)->apply(SZ, 0, DIM);
-      fld->createView("values/y")->attachBuffer(buf)->apply(SZ, 1, DIM);
-      fld->createView("values/z")->attachBuffer(buf)->apply(SZ, 2, DIM);
-      break;
-    case 2:
-      fld->createView("values/x")->attachBuffer(buf)->apply(SZ, 0, DIM);
-      fld->createView("values/y")->attachBuffer(buf)->apply(SZ, 1, DIM);
-      break;
-    default:
-      fld->createView("values/x")->attachBuffer(buf)->apply(SZ, 0, DIM);
-      break;
+      // create views into a shared buffer for the coordinates, with stride NDIMS
```

Comment:
Minor (pre-existing typo): I think this should be `DIM` rather than `NDIMS`

```suggestion
      // create views into a shared buffer for the coordinates, with stride DIM
```